core/server/web/api/canary/admin/routes.js

```diff
@@ -17,6 +17,41 @@ module.exports = function apiRoutes() {
 
     const http = apiCanary.http;
 
+    (function ghostAuthModule(router) {
+        const settings = require('../../../../services/settings/cache');
+        const urlUtils = require('../../../../lib/url-utils');
+        const jwt = require('jsonwebtoken');
+        const jose = require('node-jose');
+        const issuer = urlUtils.urlFor('admin', true);
+
+        const dangerousPrivateKey = settings.get('ghost_private_key');
+        const keyStore = jose.JWK.createKeyStore();
+        const keyStoreReady = keyStore.add(dangerousPrivateKey, 'pem');
+
+        const getKeyID = async () => {
+            const key = await keyStoreReady;
+            return key.kid;
+        };
+
+        const sign = async (claims, options) => {
+            const kid = await getKeyID();
+            return jwt.sign(claims, dangerousPrivateKey, Object.assign({
+                issuer,
+                expiresIn: '30m',
+                audience: '@TODO@',
+                algorithm: 'RS256',
+                keyid: kid
+            }, options));
+        };
+
+        router.post('/ghost-identity', mw.authAdminApi, async (req, res) => {
```

Comment:
@allouis why is this endpoint a `POST` type instead of `GET`? We are not meant to be sending any payload here and only are interested in retrieving a token based on whoever made the request, right?

Reply:
Idea was that we are _creating_ a token for each req! But I think this would work equally well with GET. Though I wonder if it's a concern that you could be sent a URL which when clicked would generate a token? :thinking:

Reply:
> Though I wonder if it's a concern that you could be sent a URL which when clicked would generate a token?

This endpoint is meant to be used only by authenticated users, don't think miss-clicking on the URL is a concern. Or are you worried about phishing scenario of some kind?

Reply:
Yeah was thinking more being sent the link and being socially engineered to share the token :man_shrugging:

Reply:
This seems like a very unlikely scenario tbh. I'd rather follow the convention we already have for `GET`ing members signing token and just have it available on `GET` request :) But it's not a super strong argument, just don't see a reason to be super paranoid about it.

Reply:
Sure thing :relaxed: